.github/actions/setup-poetry/action.yml

```
@@ -0,0 +1,19 @@
+name: 'Set up Poetry and install'
+description: 'Set up a specific version of Poetry and install dependencies using caching.'
+inputs:
+  python-version:
+    description: "Version range or exact version of Python or PyPy to use, using SemVer's version range syntax."
+    default: '3.10'
```

I would suggest
```suggestion
    default: '3.11'
```
for aligning with our [min version](https://github.com/DS4SD/docling/blob/67a5832caedfcdd9f0c900dfb6c286888334d85b/pyproject.toml#L24)